### ISSUE TITLE
Centralized all development documentation

### ISSUE THREAD

**Issue by matklad** [near/nearcore#7670]
TL;DR: https://near.github.io/nearcore/

At the moment, the docs targeting developers of nearcore are spread out across many different locations: 

* docs/ folder in the repo
* wiki: https://github.com/near/nearcore/wiki
* the other wiki: https://wiki.near.org/contribute/contribute-overview
* various "how near works" videos on youtube
* confluence

We want to centralize all that into a single place, such that it's easier to find docs, as well as to add them (you don't need to deliberate where to put stuff). 

We discussed that a bit [on zulip](https://near.zulipchat.com/#narrow/stream/308695-pagoda.2Fprivate/topic/open.20confluence.20for.20NEAR.20docs) and on the all-hands. 

We figured out we want the following features from our doc system: 

* publicly accessible, everyone can read the docs
* publicly editable, everyone can edit docs
* comments, so that it's easy to engage with the docs without necessary editing them
* stored in the repo, so that we can give public attribution to folks contributing to docs
* stored in markdown, so that it's easy to edit in Emacs
* stored in wyswyg, so that it's easy to edit outside of Emacs
* drag'n'drop experience for media like images, videos, etc
* has table of contents and in general easy way to organize large volume of loosely connected stuff
* reasonable eye-candy in terms of CSS

We considered a bunch of options, and ended up choosing https://rust-lang.github.io/mdBook/, the same thing that rust ecosystem itself is using, eg here https://rustc-dev-guide.rust-lang.org. 

It doesn't have wyswyg, but the edit button which opens GitHub editor is pretty slick. It also doesn't support drag'n'drop for images, but an OK workflow is to drag images into a special GitHub issue, and essentially use that as an image hosting platform. It doesn't have comments, which is unfortunate. On all other accounts, it is pretty great. 

Here's the partial laundry list of things to do to finish transition: 

- [x] setup minimal github action and github pages for rendering (https://github.com/near/nearcore/pull/7668)
- [x] setup skeleton docs (https://github.com/near/nearcore/pull/7675)
- [x] move contents form repo's docs folder into book (https://github.com/near/nearcore/pull/7677)
- [x] double check source layout, naming, urls
- [x] move repo's wiki into the book, kill the wiki (https://github.com/near/nearcore/pull/7706)
- [x] move and kill the other wiki (https://github.com/near/wiki/pull/55)
- [x] make sure that contributing md and such refer to the book
- [ ] move over contents from confluence (https://github.com/near/nearcore/pull/7824, https://github.com/near/nearcore/pull/7826, https://github.com/near/nearcore/pull/7850)
- [ ] kill private pages on confluence
- [ ] kill public pages on confnuence
- [x] add "how to add new contents to mdbook" practice
- [ ] re-read all the docs, remove/rewrite obsolete & poorly written parts
- [ ] go through go links
- [ ] go through any extra content, like the other wiki, youtube and what not
- [ ] move "observability" stuff out of architeceture&style into a dedicated chapter https://github.com/near/nearcore/issues/7806
- [ ] move "testing" stuff out of architercute into the testing chapter
- [ ] refresh codemap
- [ ] re-do image which starts architecture? 
- [ ] image for the core data model (blocks, chunks, receipts, parts, etc)
- [x] link media issues from somewhere https://github.com/near/nearcore/issues/7821
- [ ] update "How neard works when it is synchronizing" in "how neard works", it's a bit obsolete

Potential future work:

- [x] https://github.com/near/nearcore/issues/7836
- [x] https://github.com/near/nearcore/issues/7827
- [ ] sort out golings -- they currently are accessible only to Pagoda employees.
- [ ] setup mdBool linkcheck
- [ ] To the architecture, add a "what" page which thoroughly documents the kinds  of data in our system -- blocks, chunks, parts, approvals, trasactions, receipts, shards, trie nodes, contracts, accounts, etc.

**Comment by akhi3030** [comment]
@matklad : I would like to volunteer to help out here.  Please assign tasks to me that you think I can help with.

**Comment by matklad** [comment]
![image](https://user-images.githubusercontent.com/1711539/198282472-3883dcc1-77ca-452c-b21e-0a7af1435ede.png)

![image](https://user-images.githubusercontent.com/1711539/198282518-cdeb375e-8f1c-4634-842c-6490020ad9c0.png)

![image](https://user-images.githubusercontent.com/1711539/198282561-c97235a1-93a1-4dc8-b6bc-ee9983376b2c.png)

![image](https://user-images.githubusercontent.com/1711539/198282601-383977f1-08dd-45fe-aa19-70556d585034.png)

![image](https://user-images.githubusercontent.com/1711539/198282641-1e728088-6f2b-4cb9-90c9-5eb09304e72a.png)
